core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -248,15 +254,71 @@ final class GBQFlow[F[_]: Concurrent](
               .as(DestinationError.invalidConfiguration(
                 (GBQDestinationModule.destinationType, 
                   config.sanitizedJson,
-                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[Uri])
-
-            
+                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[(Uri, GBQJob)])
         }
       }
     })
   }
 
-  private def upload(bytes: Stream[F, Byte], uploadLocation: F[Either[InitializationError[Json], Uri]]): Stream[F, Unit] = {
+  private def getJob(job: GBQJob): Stream[F, GBQJob] = {
+    val jobUrlString =
+      s"https://bigquery.googleapis.com/bigquery/v2/projects/${job.jobReference.projectId}/jobs/${job.id}"
+
+    val jobReq = credsF map { creds =>
+      Request[F](Method.GET, Uri.unsafeFromString(jobUrlString))
+        .withHeaders(creds)
+        .withContentType(`Content-Type`(MediaType.application.json))
+    }
+
+    Stream
+      .eval(jobReq)
+      .flatMap(req => client.stream(req))
+      .evalMap { resp =>
+        resp.status match {
+          case Status.Ok => resp.as[GBQJob]
+          case _         => ApplicativeError[F, Throwable].raiseError[GBQJob](
+                              new RuntimeException(s"getJobStatus failed: ${resp.status.reason}. Check job manually with jobId: ${job.id}" )
+                            )
+        }
+      }
+  }
+
+  private def checkJobStatus(job: GBQJob, timeOut: FiniteDuration, maxRetries: Int, tries: Int): Stream[F, Unit] = {
```

Comment:
I don't think this method needs to operate in `Stream` either and can be just `F`. I feel like maybe you chose to use `Stream` because of a downstream need to have a `Stream`. I'm not sure if I agree with that approach, I think it's better to use the weakest possible constraint for a given piece of code as a default. In this method I don't know if I see a compelling reason to use `Streams`, so imo it'd be better (and easier) to just use `F` directly